File: src/main/java/Developer.java
import java.util.List;
import java.util.LinkedList;

public class Developer extends AbstractEmployee {
    private List<Task> tasks = new LinkedList<>();
    public Developer(String name, String role) throws IllegalArgumentException
    {
        super(name, role);
    }

    public void assaign(Task task)
    {
        tasks.add(task);
        System.out.println(task);
        task.setPerformer(this);
    }

    public Report reportWork()
    {
        return new Report(tasks);
    }
}

File: src/main/scala/Developer.scala
import Gender.Gender

class Developer(firstName: String,
                lastName: String,
                role: String,
                email: Email,
                gender: Gender,
                country: String,
                university: String)
  extends AbstractEmployee(firstName, lastName, role, email, gender, country, university) {

  private var tasks_ : List[Task] = List()

  def tasks: List[Task] = tasks_

  override def reportWork: Report = new Report(tasks)

  override def assign(task: Task): Unit = {tasks_ = task.setEmployee(this) :: tasks_}
}